src/PowerShellEditorServices.Engine/Services/PowerShellContext/Handlers/EvaluateHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices;
+
+namespace PowerShellEditorServices.Engine.Services.Handlers
+{
+    public class EvaluateHandler : IEvaluateHandler
+    {
+        private readonly ILogger _logger;
+        private readonly PowerShellContextService _powerShellContextService;
+
+        public EvaluateHandler(ILoggerFactory factory, PowerShellContextService powerShellContextService)
```

Comment:
Does a handler for our own thing require a document selector?

Reply:
Nope because we control when it gets fired